src/glfw3.jl

```diff
@@ -289,12 +292,36 @@ immutable VidMode
 	refreshrate::Cint   # The refresh rate, in Hz, of the video mode.
 end
 
+immutable GLFWError <: Exception
+	code::ErrorCode
+	description::String
+end
+Base.showerror(io::IO, e::GLFWError) = print(io, "GLFWError ($(e.code)): ", e.description)
+
 #************************************************************************
 # GLFW API functions
 #************************************************************************
 
 # Initialization and version information
-Init() = Bool(ccall( (:glfwInit, lib), Cint, ())) || error("initialization failed")
+function Init()
+	try
+		return Bool(ccall( (:glfwInit, lib), Cint, ())) || error("initialization failed")
+	catch ex
+		if isa(ex, GLFWError) && ex.code == PLATFORM_ERROR && (
+				contains(ex.description, "Failed to get display service port iterator") ||
+				contains(ex.description, "Failed to retrieve display name") ||
+				contains(ex.description, "RandR gamma ramp support seems broken") ||
```

Comment:
Can you remove them? I have already `display invisible characters` turned on and can't see them. Or do you mean ex.description?